sherpa/models/regrid.py

```diff
@@ -586,6 +586,21 @@ def eval_non_integrated(self, pars, modelfunc, data_space, eval_space,
 
         return y
 
+    def eval_integrated(self, pars, modelfunc, data_space, eval_space,
+                        **kwargs):
+        """Evaluate a grid with low and high bins.
+
+        Note that data_space and eval_space are assumed to have two
+        elements (low- and high- edges).
```

Comment:
This function is called with `data_space.grid`, yet the parameter is simply called `data_space`. I think that's confusing. I would suggest to actually pass in the dataspace itself (which is closer to how it was before this was separated to its own function) to avoid that confusion.

Reply:
An alternative is to rename these parameter names. The reason I suggest this is that if you look at `eval_non_integrated` (which isn't changed in this PR) it is also sent the "actual" grid - via `data_space.midpoint_grid` - which makes it clear at the `_evaluate` call sight that it is different to the integrated case.  

So I propose

```
    def eval_non_integrated(self, pars, modelfunc, data_grid, eval_grid,
                            **kwargs):

    def eval_integrated(self, pars, modelfunc, data_grid, eval_grid,
                        **kwargs):
```

How does that sound?

Reply:
And improving the docstrings for these these methods...

Reply:
That's a good solution, too.

Reply:
I've rebased this PR to use add docstrings and use a more-consistent naming scheme for the parameters